### DISCUSSION TITLE
Peacock terminated suddenly

### DISCUSSION THREAD

**Discussion by MAbdoelatef** [idaholab/moose#16590] [Q&A Getting Started] [answered]
Merry Christmas,
Can you please advise why Peacock is terminated suddenly (see the attached message)? It might be a Mac issue, not the MOOSE issue, but I still couldn't know why!. 
Many thanks 
<img width="502" alt="Screen Shot 2020-12-25 at 5 38 37 PM" src="https://user-images.githubusercontent.com/43039138/103143394-a0d41300-46da-11eb-8dc0-e84b5f87a57a.png">

**Comment by aeslaughter** [comment]
This seems like a problem with python, what version of python are you using and how was it installed? What does the command `which python` return?

**Comment by MAbdoelatef** [comment]
Many thanks, Andrew, I believe it is Python 3.8 (see the attached
screenshot), it was installed by default while installing MOOSE by conda.
[image: Screen Shot 2021-01-04 at 10.13.26 AM.png]

On Mon, Jan 4, 2021 at 9:31 AM Andrew E Slaughter <notifications@github.com>
wrote:

> This seems like a problem with python, what version of python are you
> using and how was it installed? What does the command which python return?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/idaholab/moose/discussions/16590*discussioncomment-260280__;Iw!!KwNVnqRv!TSZXn_hbiT8t2u3Oxv7K3UTdu_pWQC5ytUnq1DxWidpIxWKr8tff24pogx000t-8aKQ$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AKILTIQAJ2GB6TMB5A6FSNDSYHNLHANCNFSM4VJOSYZA__;!!KwNVnqRv!TSZXn_hbiT8t2u3Oxv7K3UTdu_pWQC5ytUnq1DxWidpIxWKr8tff24pogx00_vJvjmY$>
> .
>


-- 
*Sincerely,*
*Mohammed Abdoelatef  | * Graduate Assistant Research
[image: Materials Development and characterization Center(MDC 2)]
Mobile: 979.587.0877 | M.Abdoelatef@tamu.edu
- - - - - - - - - - - - - - - - - - - - - - - -
TEXAS A&M UNIVERSITY | FEARLESS on Every Front

**Reply by aeslaughter** [reply]
If you have time give it a try with python 3.7.

```
conda create -n moose-py37 python=3.7 moose-tools moose-libmesh
conda activate moose-py37
```

**Reply by aeslaughter** [reply]
Nevermind, it is 3.7 already. It is in the error message. I missed that. However, the path is not complete, it could be that it is picking up some other python you have installed.

**Comment by MAbdoelatef** [comment]
Then, should I remove all the python versions I have, then reinstall again.
If yes, what are the best commands to do that.

On Mon, Jan 4, 2021 at 10:24 AM Andrew E Slaughter <notifications@github.com>
wrote:

> Nevermind, it is 3.7 already. It is in the error message. I missed that.
> However, the path is not complete, it could be that it is picking up some
> other python you have installed.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/idaholab/moose/discussions/16590*discussioncomment-260425__;Iw!!KwNVnqRv!Tx9KEzgJHB0svaYWxuWXRd1-NBsV44Fggjd3nvgn_dWLw5_auW7kH8X-DEKAdNqByEs$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AKILTIXF2RTFZFRKLUNOZ2DSYHTVFANCNFSM4VJOSYZA__;!!KwNVnqRv!Tx9KEzgJHB0svaYWxuWXRd1-NBsV44Fggjd3nvgn_dWLw5_auW7kH8X-DEKA3tIVbEY$>
> .
>


-- 
*Sincerely,*
*Mohammed Abdoelatef  | * Graduate Assistant Research
[image: Materials Development and characterization Center(MDC 2)]
Mobile: 979.587.0877 | M.Abdoelatef@tamu.edu
- - - - - - - - - - - - - - - - - - - - - - - -
TEXAS A&M UNIVERSITY | FEARLESS on Every Front

**Reply by aeslaughter** [reply]
I am just trying to figure out what python is being used. If you type `which python` in the same window you are running peacock, what do you get?

**Comment by MAbdoelatef** [comment, marked as answer]
I think the same window is not allowing me to add any other command, see
below. Am I missing something?
[image: Screen Shot 2021-01-04 at 2.06.12 PM.png]

On Mon, Jan 4, 2021 at 11:18 AM Andrew E Slaughter <notifications@github.com>
wrote:

> I am just trying to figure out what python is being used. If you type which
> python in the same window you are running peacock, what do you get?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/idaholab/moose/discussions/16590*discussioncomment-260583__;Iw!!KwNVnqRv!S_RXFUrF3zi0QZISjztC0nSiqNsbfcgnJnbDi8_FzD8pj-fnr0-AYR3Z9G_LVCxaTtM$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AKILTIVIRTV5K2Q3VDPZ3E3SYHZ6ZANCNFSM4VJOSYZA__;!!KwNVnqRv!S_RXFUrF3zi0QZISjztC0nSiqNsbfcgnJnbDi8_FzD8pj-fnr0-AYR3Z9G_LpWdSLdI$>
> .
>


-- 
*Sincerely,*
*Mohammed Abdoelatef  | * Graduate Assistant Research
[image: Materials Development and characterization Center(MDC 2)]
Mobile: 979.587.0877 | M.Abdoelatef@tamu.edu
- - - - - - - - - - - - - - - - - - - - - - - -
TEXAS A&M UNIVERSITY | FEARLESS on Every Front

**Reply by aeslaughter** [reply]
The image isn't showing up.

**Reply by aeslaughter** [reply]
I think you will need to go to the discussions thread to add the image, the attachment via the reply email is not working.

**Reply by MAbdoelatef** [reply]
<img width="1566" alt="Screen Shot 2021-01-04 at 2 06 12 PM" src="https://user-images.githubusercontent.com/43039138/103578408-1e96dc00-4e9c-11eb-82ba-538cfeb57f9a.png">

**Reply by aeslaughter** [reply]
That is what I would expect for python, coming from miniconda3 location. I am not sure why you are getting the MacOS failure, but the real problem seems to be in your input file which is producing an error as shown in this screen shot.

**Reply by MAbdoelatef** [reply]
Actually, the peacock problem happens for both working and non-working input files. I believe the non-working input file is not the reason. I will try to remove the python and reinstall it again; I will notify you if it works. 
Many thanks for taking the time to consider this issue.

**Reply by aeslaughter** [reply]
Can you consistently reproduce the MacOS "python quit unexpectedly" problem? If so, what are the steps for doing that.

**Comment by MAbdoelatef** [comment]
Please see the attached screenshot.

On Mon, Jan 4, 2021 at 2:17 PM Andrew E Slaughter <notifications@github.com>
wrote:

> The image isn't showing up.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/idaholab/moose/discussions/16590*discussioncomment-261079__;Iw!!KwNVnqRv!UqJa0P7h1mCRHWjwtDh3aaeSxzPCwXAbANQA_rhg2RPfvTuvXV8L4NxJxMjyONYLKqU$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AKILTIRVFIUOVOQKA5YAXR3SYIO7HANCNFSM4VJOSYZA__;!!KwNVnqRv!UqJa0P7h1mCRHWjwtDh3aaeSxzPCwXAbANQA_rhg2RPfvTuvXV8L4NxJxMjyXnfEMWo$>
> .
>


-- 
*Sincerely,*
*Mohammed Abdoelatef  | * Graduate Assistant Research
[image: Materials Development and characterization Center(MDC 2)]
Mobile: 979.587.0877 | M.Abdoelatef@tamu.edu
- - - - - - - - - - - - - - - - - - - - - - - -
TEXAS A&M UNIVERSITY | FEARLESS on Every Front

**Reply by friedmud** [reply]
Just before running the peacock command - please do `which python`

**Reply by aeslaughter** [reply]
That is in the screenshot above, in the upper left terminal. It looks correct.

**Reply by friedmud** [reply]
Sorry @aeslaughter you're right - I only saw the one he did after he started Peacock
